File: Java_Programming/Practise_Programs/PP_106.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_106
//
//  Description       : This program demonstrates accessing a
//                      non-static function of another class by
//                      creating an object of that class.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 12/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

class ArrayX
{
//////////////////////////////////////////////////////////////////
//
//  Function Name     : Update()
//
//  Description       : It is used to update each element of the
//                      array by incrementing its value by 1.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 12/08/2026
//
//////////////////////////////////////////////////////////////////

    public void Update(int Arr[])
    {
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt]++;       
        }
    }
}

class PP_106
{
    public static void main(String A[])
    {
        int Brr[] = {10,20,30,40,50};
        int iCnt = 0;

        System.out.println("Array elements before function call : ");

        for(iCnt = 0; iCnt < 5; iCnt++)
        {
            System.out.println(Brr[iCnt]);
        }

        ArrayX aobj = new ArrayX();     
        aobj.Update(Brr);

        System.out.println("Array elements before function call : ");

        for(iCnt = 0; iCnt < 5; iCnt++)
        {
            System.out.println(Brr[iCnt]);
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Array elements before function call :
//      10
//      20
//      30
//      40
//      50
//
//      Array elements before function call :
//      11
//      21
//      31
//      41
//      51
//
//////////////////////////////////////////////////////////////////

File: C_Programming/Practise_Programs/PP_106.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_106
//
//  Description       : This program calculates the length of a
//                      string using a character pointer and while loop.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    char *str = "Ganesh";    
    int iCount = 0;  

    while(*str != '\0')
    {
        iCount++;
        str++;
    }    

    printf("Length of string is : %d\n", iCount);
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Length of string is : 6
//
//////////////////////////////////////////////////////////////////